### PR TITLE
Bug 1880103: Bump deps for Kube 1.19

### PR DESCRIPTION
**Pull request by sgreene570**
This PR bumps the kube specific dependencies in `openshift/coredns` to kube 1.19. 

The following upstream PRs were (paritally) cherry-picked to accommodate this.

* https://github.com/coredns/coredns/pull/3552
* https://github.com/coredns/coredns/pull/3794
* https://github.com/coredns/coredns/pull/3796

### CODE REVIEW

**Comment by sgreene570** [comment]
Unit test failures are `go1.12` specific.
Depends on https://github.com/openshift/coredns/pull/41
/hold

**Comment by sgreene570** [comment]
needs https://github.com/openshift/release/pull/12084

File: core/plugin/zplugin.go [hunk 1]
```diff
@@ -48,6 +48,4 @@ import (
 	_ "github.com/coredns/coredns/plugin/trace"
 	_ "github.com/coredns/coredns/plugin/transfer"
 	_ "github.com/coredns/coredns/plugin/whoami"
-	_ "github.com/coredns/federation"
-	_ "github.com/openshift/coredns-mdns"
```
Comment (Miciah) [inline core/plugin/zplugin.go:52]: We need to keep coredns-mdns for KNI (bare metal).

**Review by Miciah — CHANGES_REQUESTED** [review]

File: core/plugin/zplugin.go (hunk 1 above)
Comment (sgreene570) [reply core/plugin/zplugin.go:52]: Whoops. Pushing a fix. Thanks!

Comment (sgreene570) [reply core/plugin/zplugin.go:52]: (Note to self: Triple check upstream commit contents before cherry-picking)

Comment (sgreene570) [reply core/plugin/zplugin.go:52]: Fixed :grin:

**Comment by cybertron** [comment]
We need https://github.com/openshift/release/pull/9770 to have the metal ci job on this repo. I don't expect problems from this change, but we should test it to be sure.

**Comment by sgreene570** [comment]
@cybertron looks like the metal-ipi job passed. Thanks for the input!

**Pull request merged**